webviz_config/utils/_callback_typecheck.py

```diff
@@ -9,12 +9,44 @@ class ConversionError(Exception):
     pass
 
 
+def _isinstance(arg: Any, annotation: Any) -> bool:
+    # pylint: disable=too-many-return-statements
+    if annotation is None:
+        return arg is None
+
+    if get_origin(annotation) is None:
+        try:
+            return isinstance(arg, annotation)
+        except TypeError:
+            return False
+
+    if get_origin(annotation) == Union:
+        for annotation_arg in get_args(annotation):
+            if _isinstance(arg, annotation_arg):
+                return True
+
+    if get_origin(annotation) is list and isinstance(arg, list):
+        result = True
+        for annotation_arg in arg:
+            result &= _isinstance(annotation_arg, get_args(annotation)[0])
+        return result
+
+    if get_origin(annotation) is dict and isinstance(arg, dict):
+        result = True
+        for key, value in arg.items():
+            result &= _isinstance(key, get_args(annotation)[0])
+            result &= _isinstance(value, get_args(annotation)[1])
```

Comment:
The `&=` syntax was new to me. What does it do here? And is the `result` assigned on line 37 used somewhere?

Reply:
`&=` is a `bitwise and` - it compares two arrays of bits and returns a new one holding `1` where both array values were `1` and `0` where either one or both of them were `0`.

```python
a = 00000101
b = 00000011

a & b = 00000001

a = 1 # True
b = 0 # False

a & b = 0

a = 1 # True
b = 1 # True

a & b = 1
```

One could also write
```python
result = result and _isinstance(key, get_args(annotation)[0])
```